src/main/java/com/aws/greengrass/mqttclient/AwsIotMqtt5Client.java

```diff
@@ -380,7 +389,7 @@ public void close() {
     public void reconnect(long timeoutMs) throws TimeoutException, ExecutionException, InterruptedException {
         logger.atInfo().log("Reconnecting MQTT client most likely due to device configuration change");
         disconnect().get(timeoutMs, TimeUnit.MILLISECONDS);
-        connect().get(timeoutMs, TimeUnit.MILLISECONDS);
+        connect(true).get(timeoutMs, TimeUnit.MILLISECONDS);
```

Comment:
You don't actually know that we ever successfully connected at this point. Store an atomic boolean to represent if we've ever successfully connected which is set in the successful connection callback.
Then read the flag when connecting.

Reply:
ah right, looks like theres one that exists already, will rework